src/api/macros.rs

```diff
@@ -12,6 +13,7 @@ macro_rules! generate_enums {
 
     #[derive(Clone, Eq, PartialEq, Debug)]
     #[allow(clippy::large_enum_variant)]
+    #[non_exhaustive]
```

Comment:
I think the `impl_request` and  `impl_reply` should also be marked `non_exhaustive` right?

Reply:
Hmm, I’m not sure.  Without the fundamental changes to the requests that you suggested, I think adding fields to `Request` or `Response` should be breaking.